cfme/cloud/instance/__init__.py

```diff
@@ -1,80 +1,72 @@
-""" A model of Instances page in CFME."""
-from functools import partial
+from riggerlib import recursive_update
 
 from navmazing import NavigateToSibling, NavigateToAttribute
-from selenium.common.exceptions import NoSuchElementException
+from widgetastic.exceptions import NoSuchElementException
+from widgetastic.widget import View, Text
+from widgetastic_patternfly import Accordion, Dropdown, Button, Input, BootstrapSelect
+from widgetastic_manageiq import (
+    ManageIQTree, Table, CheckboxSelect, Calendar, PaginationPane, TimelinesView, SummaryTable,
+    ItemsToolBarViewSelector)
 
 from cfme import BaseLoggedInPage
 from cfme.common.vm import VM
-from cfme.exceptions import (
-    InstanceNotFound, OptionNotAvailable, DestinationNotFound,
-    BlockTypeUnknown, TemplateNotFound, ToolbarOptionGreyedOrUnavailable)
+from cfme.exceptions import InstanceNotFound, DestinationNotFound, TemplateNotFound
 from cfme.fixtures import pytest_selenium as sel
-from cfme.web_ui import (
-    accordion, paginator, toolbar as tb, CheckboxTree, Region, InfoBlock,
-    Tree, Quadicon, match_location, Form, Table, PagedTable, form_buttons)
+from cfme.provisioning import ProvisioningForm
+from cfme.web_ui import flash, Quadicon
 from cfme.web_ui.search import search_box
 from utils import version
 from utils.appliance import Navigatable
 from utils.appliance.implementations.ui import navigate_to, CFMENavigateStep, navigator
-from utils.wait import wait_for
 from utils.log import logger
-from widgetastic_manageiq import TimelinesView
-
-
-cfg_btn = partial(tb.select, 'Configuration')
-pwr_btn = partial(tb.select, 'Power')
-life_btn = partial(tb.select, 'Lifecycle')
-pol_btn = partial(tb.select, 'Policy')
-mon_btn = partial(tb.select, 'Monitoring')
-
-tree_inst_by_prov = partial(accordion.tree, "Instances by Provider")
-tree_instances = partial(accordion.tree, "Instances")
-tree_image_by_prov = partial(accordion.tree, "Images by Provider")
-tree_images = partial(accordion.tree, "Images")
-
-list_page = Region(title='Instances')
-
-policy_page = Region(
-    locators={
-        'policy_tree': Tree('//div[@class="containerTableStyle"]/table')
-    })
-
-image_select_form = Form(
-    fields=[
-        ('template_table', Table('//div[@id="pre_prov_div"]//table')),
-        ('cancel_button', form_buttons.cancel)
-    ]
-)
+from utils.wait import wait_for
 
-manage_policies_tree = CheckboxTree("//div[@id='protect_treebox']/ul")
 
-list_table = PagedTable(table_locator="//div[@id='list_grid']//table")
+class CloudInstanceView(BaseLoggedInPage):
```

Comment:
@izapolsk I would like to make this refactor in conjunction with the common views for virtual machines and instances.